apps/api/v2/src/ee/calendars/services/gcal.service.ts

```diff
@@ -149,10 +144,30 @@ export class GoogleCalendarService implements OAuthCalendarApp {
     const primaryCal = cals.data.items?.find((cal) => cal.primary);
 
     if (primaryCal?.id) {
-      await this.selectedCalendarsRepository.createSelectedCalendar(
-        primaryCal.id,
-        credential.id,
+      const alreadyExistingSelectedCalendar = await this.selectedCalendarsRepository.getUserSelectedCalendar(
```

Comment:
Question: i see this is within `if (primaryCal?.id)` - does this mean we only check if there is primary calendar? Could it be that the gcal is not primary calendar but its connected?

Reply:
Both google and outlook set a default primary calendar at the time of integration so we're sure here that there will definitely be a primary calendar. This is the not the case with apple calendar thats why we need to check it against the credential